types/react-dom/index.d.ts

```diff
@@ -21,29 +21,44 @@ export function findDOMNode(instance: ReactInstance): Element;
 export function render<P extends DOMAttributes<T>, T extends Element>(
     element: DOMElement<P, T>,
     container: Element | null,
-    callback?: (element: T) => any
+    callback?: () => void
 ): T;
-export function render<P>(
-    element: SFCElement<P>,
+
+export function render(
+    element: Array<DOMElement<any, any>>,
+    container: Element | null,
+    callback?: () => void
+): Element;
+
+export function render(
+    element: SFCElement<any> | Array<SFCElement<any>>,
     container: Element | null,
     callback?: () => any
 ): void;
+
 export function render<P, T extends Component<P, ComponentState>>(
     element: CElement<P, T>,
     container: Element | null,
-    callback?: (component: T) => any
+    callback?: () => void
 ): T;
+
+export function render(
+    element: Array<CElement<any, Component<any, ComponentState>>>,
+    container: Element | null,
+    callback?: () => void
+): Component<any, ComponentState>;
```

Comment:
should this be an array return value?

Reply:
I don't think so. It seems that React-Fiber does not return the whole array back which is somewhat strange. According to my tests, it only returns the first element of the array.

Reply:
Returning something from `render` is itself deprecated. Function `ref`s should be used.

I'd be in favor of declaring this as `void` even though it's technically _de facto_ wrong.

Reply:
I would leave this as it is for now. When it's finally changed we can reflect the changes afterwards. It may be deprecated but some users might still use this in their code. I guess it's better this way.

Reply:
At least adding a comment with a deprecated tag should be enough in my opinion.

Reply:
Where would you like to add them? Marking the whole render function as deprecated doesn't feel right (Using jsdoc's @deprecated). Or do you think a simple comment would be sufficient?

Reply:
I thought something with function overload. But other than that, simple comment should enough.

Reply:
I've pushed a new commit which adds a deprecated comment.